### PR TITLE
Updated the CKEditor 4 migration guide

### LINKED ISSUE #1410: [Docs] Update the migration guide
This page became outdated after the last releases: https://ckeditor.com/docs/ckeditor5/latest/builds/guides/migrate.html

For example paste from word is mentioned there as "planned", "Media embedding is not supported yet, but it is planned." and so on.

### PR DESCRIPTION
**Pull request by AnnaTomanek**
Docs: Updated the CKEditor 4 migration guide. Closes #1410.

---

### Additional information

* Added missing CKEditor 4 options that were added in recent months.
* Reviewed existing content, cross-linked to feature guides and API docs.

### CODE REVIEW

**Comment by mlewand** [comment]
Pushed a minor typo fix, be sure to pull changes before modifying.

File: docs/builds/guides/migrate.md [hunk 1]
```diff
@@ -253,25 +285,36 @@ Note: The number of options was reduced on purpose. We understood that configuri
...
 			<td>N/A</td>
 		</tr>
 		<tr>
 			<td><a href="/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-embed_provider">embed_provider</a></td>
-			<td>N/A</td>
+			<td>Refer to the {@link features/media-embed Media embed feature} guide to learn more about media embedding in CKEditor 5.</td>
+		</tr>
+		<tr>
+			<td><a href="/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-emoji_emojiListUrl">emoji_emojiListUrl</a> <br> <a href="/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-emoji_minChars">emoji_minChars</a></td>
+			<td>Emoji can be pasted into CKEditor 5 as Unicode content.</td>
```
[the remark this answers is not in the document]
Comment (Reinmar) [reply docs/builds/guides/migrate.md]: I would say that there's emoji support. You just need to use your native OS's capabilities (such as Cmd+Ctr+Space on macOS). We can actually list those keystrokes.

**Comment by Reinmar** [comment]
I left feedback in the original ticket about the things that were unclear.

**Comment by Reinmar** [comment]
Lemme know if more feedback is needed from me.

**Comment by Reinmar** [comment]
On it.

**Comment by Reinmar** [comment]
OK, I've read pretty much everything. Too much to be 100% thorough, but I caught quite a lot of things still so I guess I was thorough enough :D 

Merging to master ❤️

**Pull request merged**